spark/src/test/scala/org/apache/spark/sql/delta/MergeIntoSuiteBase.scala

```diff
@@ -3356,6 +3357,25 @@ abstract class MergeIntoSuiteBase
     "delta.dml.merge.findTouchedFiles",
     "delta.dml.merge.writeInsertsOnlyWhenNoMatches",
     "delta.dml.merge")
+
+  test("merge execution is recorded with QueryExecutionListener") {
+    withKeyValueData(
+      source = (0, 0) :: (1, 10) :: Nil,
+      target = (1, 1) :: (2, 2) :: Nil) { case (sourceName, targetName) =>
+      val plans = withLogicalPlansCaptured(spark, optimizedPlan = false) {
+        executeMerge(
+          tgt = s"$targetName t",
+          src = s"$sourceName s",
+          cond = "s.key = t.key",
+          update(set = "*"))
+      }
+      val mergeCommand = plans.collectFirst {
+        case m: MergeIntoCommand => m
+      }
+      assert(mergeCommand.nonEmpty,
```

Comment:
```suggestion

      assert(mergeCommand.nonEmpty,
```

Reply:
I kept the code without new lines, hope you don't mind

Reply:
@johanl-db Yeah I should have mentioned that's a nit, I'm fine with it either way.